### PR TITLE
Rewrite collection with tanstack virtual, move non-page components

### PR DESCRIPTION
**Pull request by harshithmohan**

### CODE REVIEW

File: src/components/Collection/CollectionView.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+          // Here, i will be the actual index of the group in group list
+          for (let i = fromIndex; i < toIndex; i++) {
+            const neededPage = Math.ceil((i + 1) / pageSize);
+            const relativeIndex = i % pageSize;
+            const groupList = groupPages[neededPage];
+
+            if (groupList === undefined && !fetchingPage) {
+              setFetchingPage(true);
+              fetchPage(neededPage);
```
Comment (hidden4003) [inline src/components/Collection/CollectionView.tsx:121]: This does not have to be here, can be moved safely outside the for loop and improve performance and readability a bit, essentially you only need to check 2 indices, the starting one and ending one and in-between you can skip to determine pages you need to load.

File: src/components/Collection/CollectionView.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            }
+
+            const item = groupList !== undefined ? groupList[relativeIndex] : undefined;
```
Comment (hidden4003) [inline src/components/Collection/CollectionView.tsx:124]: If you are on last/not full page, doesn't it mean you will start getting empty items here?

**Review by hidden4003 — APPROVED** [review]

File: src/components/Collection/CollectionView.tsx (hunk 2 above)
Comment (harshithmohan) [reply src/components/Collection/CollectionView.tsx:124]: `groupList[relativeIndex]` will return `undefined` and then in the next line `isPlaceholder` will become true, which is needed to render empty divs.
Without the placeholder divs, the items in the last row will be center aligned as each row has `justify-center` to not have more empty on the right than the left

File: src/components/Collection/CollectionView.tsx (hunk 1 above)
Comment (harshithmohan) [reply src/components/Collection/CollectionView.tsx]: Moved it outside the loop

**Pull request merged**